#184752095 Admin should be able to disable an account

Gabriel-mdv opened the pull request:
## What does this PR do?
This PR gives authority to the admin to enable or disable an account. The disabled or enabled account owner is informed by email.

## Description of Task to be completed?

GIVEN admin and multiple users exists on the sytem
WHEN a user is suspected or proved to have violated the terms of service of the application
THEN admin is able to trigger an action to block usage of a particular account and an email sent to the user with reasons from the block.

## How should this be manually tested?

-Clone the repository using `git clone` link
-Install node packages that the application depends on: `npm install`
-Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example`  file available in root directory of the application.
-Start the application development server: `npm run dev`
-Test the user registration endpoint available on this address: `https://localhost:{PORT}/api/user/disable/{id}` to disable the user or:  `https://localhost:{PORT}/api/user/enable/{id}` to enable the user again, PORT being a local port that the application is running on.
-When the user is disabled, he cannot login

## Any background context you want to provide?
* Admin is able to disable the account
* Only the admin is able to disable the account
* When the user tries to disable the account he get the error of unauthorized -a disabled or enabled user is sent an email


## What are the relevant pivotal tracker/Trello stories?

  - [#184752095](https://www.pivotaltracker.com/story/show/184752095)

nishimweprince commented on src/utils/emails.js:
```diff
@@ -64,10 +91,11 @@ const sendEmail = async (email, name, heading, res, messageTemplate, token) => {
     // RETURN SUCCESS MESSAGE
     /* eslint-disable no-console */
-    console.log(`Email sent to ${name}`);
+    console.log(message);
```
Logging a complete would be lengthy. I would suggest to log only a confirmation that the email has been sent. Similar to what was removed on [line 67](https://github.com/atlp-rwanda/e-comm-team-techsmith-bn/pull/42/files#diff-8258c2a30dd09b705828fbe37cc2599233a09900f067b13da8beb82b80cc2fa6L67).

nishimweprince commented on src/utils/emails.js:
```diff
@@ -64,10 +91,11 @@ const sendEmail = async (email, name, heading, res, messageTemplate, token) => {
+    console.log(message);
   } catch (error) {
+    console.log(error);
     return res.status(500).json({
```
I would also suggest to not return a status when an email was not sent because it crushes the whole application when an email could not be delivered. Examples are when our SendGrid daily credits are exhausted. Instead, you can just log on console for now.

nishimweprince commented on src/controller/loginController.js:
```diff
@@ -21,6 +21,7 @@ const loginController = async (req, res) => {
     if (checkPassword) {
       const payload = {
+        role: findUser.roleId,
         id: findUser.email,
```
Consider using the user id instead of their email.

nishimweprince reviewed (changes requested):
You still have conflicts with the default branch. Rebase again to ensure you have the latest updates. I also left some comments.

jcharles921 reviewed (approved):
Great work

jcharles921 commented:
Resolve conflicts tho

niyontwali reviewed (changes requested):
Please check in your controller of dis_enableUser.js and make the following changes;

1. First and foremost, please change the name of your controller. Use camelCase in naming your controllers since it the practice other used
2. In your controller file, please use res.status().json() instead of res.status().send(). Please adhere to that practice in all your codebase
3. The middleware you created to check if a user is the admin, please check there is a middleware to verify if the user is an admin and use that one and delete yours. This is to avoid dumplicate middlewares. Note that you might need to check and see if the middleware existing is well addressed. 
4. Please adjust any other file that would be affected by the changes you are requested to adjust.

Gabriel-mdv committed:
feature(disable User): Admin should be able to desable and enable users

only the admin is able to perform this action
after the action the disabled or enable user is sent an email
a regular user is not allowed to perform this action

[delivers]  #184752095

jcharles921 reviewed (approved):
Great Job Bro 👊🏾
